lightningd/jsonrpc.c

```diff
@@ -357,27 +290,34 @@ static void destroy_command(struct command *cmd)
 	cmd->jcon->command = NULL;
 }
 
-/* FIXME: Remove result arg here! */
 void command_success(struct command *cmd, struct json_stream *result)
 {
 	assert(cmd);
 	assert(cmd->have_json_stream);
-	if (cmd->jcon)
-		jcon_append(cmd->jcon, " }\n\n");
+	json_stream_append(result, " }\n\n");
+	json_stream_close(result, cmd);
 	if (cmd->ok)
 		*(cmd->ok) = true;
+
+	/* If we have a jcon, it will free result for us. */
+	if (cmd->jcon)
+		tal_steal(cmd->jcon, result);
```

Comment:
If we do not have a `jcon`, does `result` get taken by this function?

Reply:
This function frees `cmd` a few lines below, so `result` goes with it...